cpu/cpu.go

```diff
@@ -182,26 +196,41 @@ func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.Metri
 		p.snapMetricsNames, p.procStatMetricsNames); err != nil {
 		return nil, err
 	}
+	ts := time.Now()
 	for _, metricType := range metricTypes {
 		ns := metricType.Namespace()
 		if len(ns) != maxNamespaceSize {
 			return nil, fmt.Errorf("Incorrect namespace length (len = %d)", len(ns))
 		}
-
-		val, err := getMapValueByNamespace(p.stats[ns.Strings()[3]], ns.Strings()[4:])
-
-		if err != nil {
-			return metrics, err
-		}
-
-		metric := plugin.MetricType{
-			Namespace_: ns,
-			Data_:      val,
-			Timestamp_: time.Now(),
+		if ns[len(ns)-2].Value == "*" {
+			for cpuId, cpuStats := range p.stats {
+				for k, v := range cpuStats {
+					if ns[len(ns)-1].Value == k && v != nil {
+						ns1 := make([]core.NamespaceElement, len(ns))
+						copy(ns1, ns)
+						ns1[len(ns)-2].Value = cpuId
+						metric := plugin.MetricType{
+							Namespace_: ns1,
+							Data_:      v,
+							Timestamp_: ts,
+						}
+						metrics = append(metrics, metric)
+					}
+				}
+			}
+		} else {
+			val, err := getMapValueByNamespace(p.stats[ns.Strings()[3]], ns.Strings()[4:])
+			if err != nil {
+				return metrics, err
+			}
+			metric := plugin.MetricType{
+				Namespace_: ns,
+				Data_:      val,
+				Timestamp_: ts,
```

Comment:
Consider set Version of plugin as below:
```
metric := plugin.MetricType{
  Namespace_: ns,
  Data_:      val,
  Timestamp_: ts,
  Version: version,
}
```
so after collection the metric will contain information which version of plugin was used to collect it. 
The same can be added in [line 215](https://github.com/obourdon/snap-plugin-collector-cpu/blob/dynamic-metrics/cpu/cpu.go#L215)